plugins/modules/ndo_qos_class_policy.py

```diff
@@ -0,0 +1,544 @@
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    template_id = module.params.get("template_id")
+    name = module.params.get("name")
+    description = module.params.get("description")
+    uuid = module.params.get("uuid")
+    preserve_cos = module.params.get("preserve_cos")
+    qos_levels = module.params.get("qos_levels")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_policy", template, template_id)
+    mso_template.validate_template("fabricPolicy")
+    object_description = "QoS Class Policy"
+
+    path = "/fabricPolicyTemplate/template/qosClass"
+    existing_qos_policies = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {}).get("qosClass", {})
+    if name or uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            [existing_qos_policies],
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(mso_template.update_config_with_template_and_references(match.details))
+    else:
+        mso.existing = mso.previous = mso_template.update_config_with_template_and_references(existing_qos_policies)
+
+    if state == "present":
+        mso_values = {
+            "name": name,
+            "description": description,
+            "preserveCos": preserve_cos,
+        }
+        qos_levels, remove_data = format_qos_levels(mso, qos_levels)
+        if isinstance(qos_levels, dict):
+            mso_values.update(qos_levels)
+        if match:
+            append_update_ops_data(ops, match.details, path, mso_values, remove_data)
+            mso.sanitize(match.details, collate=True)
+
+        else:
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path=path, value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path=path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        qos_policies = response.get("fabricPolicyTemplate", {}).get("template", {}).get("qosClass", {})
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            [qos_policies],
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            mso.existing = mso_template.update_config_with_template_and_references(match.details)
+        else:
+            mso.existing = {}
+    elif module.check_mode and state != "query":
+        mso_template.update_config_with_template_and_references(mso.proposed)
+        mso.existing = mso.proposed if state == "present" else {}
+
+    mso.exit_json()
+
+
+def format_qos_levels(mso, qos_levels=None):
+    remove_data = []
+    if isinstance(qos_levels, list):
+        exsting_qos_levels = set()
```

Comment:
why name it existing when this is still to be configured?

typo in the variable name

```suggestion
        existing_qos_levels = set()
```

Reply:
Changed the logic of code, I removed this local variable.